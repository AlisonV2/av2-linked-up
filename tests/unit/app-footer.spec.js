import { shallowMount } from '@vue/test-utils';
import AppFooter from '@/components/nav/Footer.vue';

const config = {
  data() {
    return {
      year: '2021',
    };
  },
};

describe('AppFooter.vue', () => {
  it('AppFooter Snapshot', () => {
    const component = shallowMount(AppFooter, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Checks year rendering', () => {
    const component = shallowMount(AppFooter, config);
    const copyright = component.get('.footer-right');
    expect(copyright.text()).toBe(
      'Â© Copyright 2021 - Alison Vandromme - All rights reserved'
    );
  });
});
