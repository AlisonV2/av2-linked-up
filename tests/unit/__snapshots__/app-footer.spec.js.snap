// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter.vue AppFooter Snapshot 1`] = `
<footer
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col-12"
    >
      <div
        class="footer-copyright"
      >
        <p
          class="copyright"
        >
          <span
            data-test="copyright"
          >
             Â© Copyright 2021 - 
            <a
              href="https://alisonvandromme.com"
              rel="noopener"
              target="_blank"
            >
              Alison Vandromme
            </a>
             - All rights reserved 
          </span>
        </p>
      </div>
    </div>
  </div>
</footer>
`;
